sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/util/RetryHttpRequestInitializer.java

```diff
@@ -73,7 +73,9 @@ public boolean handleIOException(HttpRequest request, boolean supportsRetry)
       if (willRetry) {
         LOG.debug("Request failed with IOException, will retry: {}", request.getUrl());
       } else {
-        LOG.warn("Request failed with IOException, will NOT retry: {}", request.getUrl());
+        LOG.warn(
```

Comment:
It would be good if we could log how many times we retried in the low level loop here (maybe not practical for this PR).

Also, the wording I would put is that the "caller is responsible for retry" which tells the user that someone else is responsible which automatically moves them up the stack.